baybe/parameters/numerical.py

```diff
@@ -37,43 +36,16 @@ class NumericalDiscreteParameter(DiscreteParameter):
     )
     """The values the parameter can take."""
 
-    tolerance: float = field(default=0.0)
-    """The absolute tolerance used for deciding whether a value is in range. A tolerance
-        larger than half the minimum distance between parameter values is not allowed
-        because that could cause ambiguity when inputting data points later."""
+    tolerance: float = field(validator=ge(0.0))
+    """The absolute tolerance used for deciding whether a value is considered in range.
+        A value is considered in range if its distance to the closest parameter value
+        is smaller than the specified tolerance."""
 
-    @tolerance.validator
-    def _validate_tolerance(  # noqa: DOC101, DOC103
-        self, _: Any, tolerance: float
-    ) -> None:
-        """Validate that the given tolerance is safe.
-
-        The tolerance is the allowed experimental uncertainty when
-        reading in measured values. A tolerance larger than half the minimum
-        distance between parameter values is not allowed because that could cause
-        ambiguity when inputting data points later.
-
-        Raises:
-            ValueError: If the tolerance is not safe.
-        """
-        # For zero tolerance, the only left requirement is that all parameter values
-        # are distinct, which is already ensured by the corresponding validator.
-        if tolerance == 0.0:
-            return
-
-        min_dist = np.diff(self.values).min()
-        if min_dist == (eps := np.nextafter(0, 1, dtype=DTypeFloatNumpy)):
-            raise NumericalUnderflowError(
-                f"The distance between any two parameter values must be at least "
-                f"twice the size of the used floating point resolution of {eps}."
-            )
-
-        if tolerance >= (max_tol := min_dist / 2.0):
-            raise ValueError(
-                f"Parameter '{self.name}' is initialized with tolerance {tolerance} "
-                f"but due to the given parameter values {self.values}, the specified "
-                f"tolerance must be smaller than {max_tol} to avoid ambiguity."
-            )
+    @tolerance.default
+    def default_tolerance(self) -> float:
+        """Set the tolerance to fraction of the smallest value distance."""
```

Comment:
This reads as if the `fraction` can be chosen by the user. Also, we should add the information in the docstring that this fraction is set to 0.1